### ISSUE TITLE
feat: Add specific recevier support

### ISSUE THREAD

**Issue by KaptinLin** [nikoksr/notify#345]
**Describe the solution you'd like**
The ability send message to specific recevier
```
telegramService, _ := telegram.New("your_telegram_api_token")
notify.UseServices(telegramService)

fooReceiver := telegram.Receiver(-1234567890)
barReceiver := telegram.Receiver(-0987654321)

// Send a message to fooReceiver .
_ = notify.To(fooReceiver).Send(
	context.Background(),
	"Hello",
	"The message to fooReceiver",
)
// Send another message to barReceiver 
_ = notify.To(fooReceiver).Send(
	context.Background(),
	"Hello",
	"The message to barReceiver",
)
```

**Comment by nikoksr** [comment]
Hey @KaptinLin, thanks for sharing your idea/problem.

Have you considered using separate `notify` instances? For example like this:

```go
// Notifier for receiver 1
service1, _ := telegram.New("")
service1.AddReceivers(-1234567890)

n1 := notify.NewWithServices(service1)
_ = n1.Send(context.Background(), "Hello", "The message to fooReceiver")


// Notifier for receiver 2
service2, _ := telegram.New("")
service2.AddReceivers(-9876543210)

n2 := notify.NewWithServices(service2)
_ = n2.Send(context.Background(), "Hello", "The message to barReceiver")
```

I can see that your solution would add more flexibility but I want to consider other solutions first before agreeing or declining your idea.

**Comment by KaptinLin** [comment]
@nikoksr yes, it can works. 
I was look for a way not initialize the service when send to different receivers.

**Comment by rafimuhammad01** [comment]
hi @nikoksr @KaptinLin has this feature been implemented? I think I want to start contributing by solving this issue, would you mind?

**Comment by KaptinLin** [comment]
nope, if you can contribute that would be awesome

**Comment by nikoksr** [comment]
@rafimuhammad01 welcome to the project. I appreciate your willingness to support the project! We'd all be excited for your contributions in any way shape or form.

@KaptinLin could you please clarify why this feature would be beneficial for the project? This seems like a subtle pattern change and I don't want to accept changes that unnecessarily change Notify's usage patterns. 

Also, I'm not happy with the linguistic choice here. Keeping in mind that chained methods should sound as close as possible to a actual english sentence, the signature is not logically sound. `Notify to <receiver> send <message>` is far from being gramatically correct. Something like `Notify send <message> to <receiver>` would be cleaner. Obviously this does not work, since the send method returns an error type and thus does not allow for further method chaining.

To summarize, I'd more than happily accept your guys' idea and contribution but we need a clear explanation of what problem this exactly solves, plus, the signature has to change. I'm not okay with `notify.To(...).Send(...)`.

cc @svaloumas

**Comment by nikoksr** [comment]
Closing until further notice.

**Comment by AnKlvy** [comment]
Hello! I have an idea to add RemoveRecievers function, because I want to Unsubscibe one service for particular user.
I use one instance for app (fix me if I do something wrong)
```func initNotify(db *sql.DB) (*notifier.NotifyService, error) {
	//init repo interface
	repo := database.NewNotifier(db)

	//add services
	emailSvc := mail.New("vdsfdfdgf@gmail.com", "587")
	ctx := context.Background()

	fcmSvc, err := fcm.New(ctx)

	if err != nil {
		return nil, err
	}
	notify.UseServices(emailSvc, fcmSvc)

	adapters := map[string]notifier.NotifyInterface{
		"email": emailSvc,
		"fcm":   fcmSvc,
	}

	notifyService := notifier.NewNotifyService(repo, adapters)
	return notifyService, nil
}```
and I want to make something like this:
```package notifier

import (
	"fmt"
	"github.com/AnKlvy/notifier-edunite/internal/database"
	"github.com/nikoksr/notify"
)

type NotifyInterface interface {
	notify.Notifier
	AddReceivers(receivers ...string)
	RemoveReceivers(receivers ...string)
}

type NotifyService struct {
	repo     database.Models
	notifier map[string]NotifyInterface
}

func NewNotifyService(repo database.Models, notifier map[string]NotifyInterface) *NotifyService {
	return &NotifyService{repo: repo, notifier: notifier}
}

func (n *NotifyService) Subscribe(userId string, channel string, token string) error {

	svc, ok := n.notifier[channel]
	if !ok {
		return fmt.Errorf("unknown channel %q", channel)
	}
	svc.AddReceivers(token)

	err := n.repo.Notifier.Subscribe(userId, channel, token)
	if err != nil {
		return err
	}
	return nil
}

func (n *NotifyService) Unsubscribe(userId string, channel string) error {

	svc, ok := n.notifier[channel]
	if !ok {
		return fmt.Errorf("unknown channel %q", channel)
	}
	svc.RemoveReceivers(channel)

	err := n.repo.Notifier.Unsubscribe(userId, channel)
	if err != nil {
		return err
	}
	return nil
}
```
